### PR TITLE
Marketplace: Add mobile view for Plugins plans

### LINKED ISSUE #67961: Marketplace: Plan information step for other screen sizes
Follow-up issue for #67574

See [comment](https://github.com/Automattic/wp-calypso/issues/67574#issuecomment-1249319530) for more details.

Acceptance criteria:

- [ ]  Has the same function as the desktop version for other screen sizes

### PR DESCRIPTION
**Pull request by lupus2k**
#### Proposed Changes

Following #67891, needs mobile view for plugins plans.

#### Testing Instructions

* Go to `/plugins/plans/:site` using mobile view
* Plans should look like the `After` screenshot.

#### Pre-merge Checklist

- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?

Fixes #67961

### CODE REVIEW

**Comment by lupus2k** [comment]
Updated Interval type toggler margin.

This mobile view looks correct to you, @vinimotaa ?
image

**Review by candy02058912 — COMMENTED** [review]
Code LGTM, but I when I'm testing on the live site, the plan pill has some CSS overrides making it look weird:
live site screenshot

It's showing up okay locally. Could you help double-check this?

**Comment by lupus2k** [comment]
> Code LGTM, but I when I'm testing on the live site, the plan pill has some CSS overrides making it look weird:

Thanks @candy02058912, I think this bug needs a new issue.

File: client/my-sites/plan-features/header.jsx [hunk 1]
```diff
@@ -31,13 +31,19 @@ import isSiteAutomatedTransfer from 'calypso/state/selectors/is-site-automated-t
 import { getCurrentPlan } from 'calypso/state/sites/plans/selectors';
 import { getSiteSlug } from 'calypso/state/sites/selectors';
 import { getSelectedSiteId } from 'calypso/state/ui/selectors';
+import { isMarketplaceFlow } from '../plugins/flows';
```
Comment (paulopmt1) [inline client/my-sites/plan-features/header.jsx]: Dusty pointed out on my last PR is recommended to import using the absolute path. It will also ask to change the order of the import.

```suggestion
import { isMarketplaceFlow } from 'calypso/my-sites/plugins/flows';
```

File: client/my-sites/plan-features/index.jsx [hunk 2]
```diff
@@ -82,6 +82,7 @@ import PlanFeaturesItem from './item';
 import PlanFeaturesActionsWrapper from './plan-features-action-wrapper';
 import PlanFeaturesScroller from './scroller';
 import './style.scss';
+import { isMarketplaceFlow } from '../plugins/flows';
```
Comment (paulopmt1) [inline client/my-sites/plan-features/index.jsx]: Same import note

**Review by paulopmt1 — APPROVED** [review]
Good job @lupus2k! Everything is working as expected!
I fixed my promo-section and added Dusty suggestion of absolute import in two places.

one thing I noted is if I update the screen on a mobile device I have this selector:

image


And if I resize from a desktop version, I see this new version:

image

It would be good for us to use the same code, but I don't see it as a problem for this PR.

**Review by lsl — CHANGES_REQUESTED** [review]
Can you break this up into smaller PR's? It's hard to see what you're changing here or why.

Do we actually need this? `/start/plans` already has a mobile view:

Screenshot 2022-09-23 at 14-07-55 Create a site — WordPress com

File: client/components/plans/plan-pill/index.jsx [hunk 3]
```diff
@@ -1,7 +1,11 @@
 import './style.scss';
 
 export default ( props ) => (
-	<div className={ `plan-pill${ props.isInSignup ? ' is-in-signup' : '' }` }>
+	<div
```
Comment (candy02058912) [inline client/components/plans/plan-pill/index.jsx:4]: This might look more readable if we're using the classNames package:
(Code below is not actually tested yet)

```
import classNames from 'classnames';

...

className={classNames('plan-pill', {
      'is-in-signup': props.isInSignup,
      'is-current-plan': props.isCurrentPlan
  })
}
```

File: client/my-sites/plan-features-comparison/style.scss [hunk 4]
```diff
@@ -1,13 +1,15 @@
...
 $plan-features-sidebar-width: 272px;
 
-.plan-pill.is-in-signup {
-	right: 52%;
-	transform: translate(50%);
-	border-radius: 4px;
-	background-color: var(--studio-green-5);
-	color: var(--studio-gray-80);
-	font-size: $font-body-small;
+.plan-features-comparison__content {
```
Comment (candy02058912) [inline client/my-sites/plan-features-comparison/style.scss:4]: I broke this out to another PR: https://github.com/Automattic/wp-calypso/pull/68184 since it's a quick fix to review and merge.

File: client/my-sites/plan-features/actions.jsx [hunk 5]
```diff
@@ -72,10 +64,25 @@ const PlanFeaturesActionsButton = ( {
...
+				{ translate( 'Select' ) }
+			</Button>
+		);
+	}
+
 	if ( current && ! isInSignup && planType !== PLAN_P2_FREE ) {
 		return (
 			<Button className={ classes } href={ manageHref } disabled={ ! manageHref }>
-				{ canPurchase ? translate( 'Manage plan' ) : translate( 'View plan' ) }
+				{ canPurchase ? translate( 'Manageaa plan' ) : translate( 'View plan' ) }
```
Comment (candy02058912) [inline client/my-sites/plan-features/actions.jsx:85]:
```suggestion
				{ canPurchase ? translate( 'Manage plan' ) : translate( 'View plan' ) }
```

File: client/my-sites/plan-features/actions.jsx [hunk 6]
```diff
@@ -72,10 +64,25 @@ const PlanFeaturesActionsButton = ( {
 		onUpgradeClick();
 	};
 
+	if ( isMarketplaceFlow( flowName ) ) {
+		if ( current ) {
+			return (
+				<Button className={ classes } disabled>
+					{ translate( 'Your current plan' ) }
```
Comment (lsl) [inline client/my-sites/plan-features/actions.jsx:71]: What does this have to do with mobile views? 

Can we deal with this problem after we get the buttons working?

File: client/my-sites/plan-features/index.jsx [hunk 7]
```diff
@@ -936,7 +949,7 @@ const ConnectedPlanFeatures = connect(
 		const canPurchase = ! isPaid || isCurrentUserCurrentPlanOwner( state, selectedSiteId );
 		const isLoggedInMonthlyPricing =
 			! isInSignup && ! isJetpack && kindOfPlanTypeSelector === 'interval';
-		const flowName = getCurrentFlowName( state );
+		const flowName = ownProps.flowName ? ownProps.flowName : getCurrentFlowName( state );
```
Comment (lsl) [inline client/my-sites/plan-features/index.jsx:952]: `flowName` is a stepper framework concept. Marketplace code should not touch/use `flowName` at all.

I don't actually know why `flowName` is even in here, probably shouldn't be.

Comment (candy02058912) [reply client/my-sites/plan-features/index.jsx:952]: I came to realize that after digging into some more of the code today. Originally didn't want to introduce a new prop `isInMarketplace` and wanted to reuse a prop that already existed.

If we want to revert this, we'll also need to check on the code [here](https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/plan-features-comparison/index.jsx#L457). I remember that was the reason why I introduced a fake `flowName` at my [first attempt](https://github.com/Automattic/wp-calypso/pull/67891/files#diff-d89d039d749332c2d6c1c257980ee3273b0db6689a013b93a7abf58a9e6267f1R69).

I think we should revert it and maybe just go back on using `isInMarketplace` because it might lead to some unexpected assumptions such as thinking `getCurrentFlowName` would work [here](https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/plan-features-comparison/header.jsx#L179). It's something meant to be incorporated with the stepper framework.

Comment (lupus2k) [reply client/my-sites/plan-features/index.jsx:952]: Going back to isInMarketplace instead of using the flowName sounds good and cleaner to me!

**Comment by lupus2k** [comment]
@vinimotaa's feedback: `let's try to get as close as possible to the mobile view of /start`

I will create a new PR to use `isInMarketplace` prop and changes in buttons / texts

**Comment by lupus2k** [comment]
Closed in favor of #68228
